src/contexts/AppController.tsx

```diff
@@ -0,0 +1,202 @@
+import React, {
+    createContext,
+    useContext,
+    useEffect,
+    useRef,
+} from "react";
+
+import useContextStore from "../stores/contextStore";
+import useLogFileManagerStore from "../stores/logFileManagerProxyStore";
+import useLogFileStore from "../stores/logFileStore";
+import useUiStore from "../stores/uiStore";
+import useViewStore from "../stores/viewStore";
+import {LOG_LEVEL} from "../typings/logs";
+import {DO_NOT_TIMEOUT_VALUE} from "../typings/notifications";
+import {UI_STATE} from "../typings/states";
+import {
+    CURSOR_CODE,
+    CursorType,
+} from "../typings/worker";
+import {
+    findNearestLessThanOrEqualElement,
+    isWithinBounds,
+} from "../utils/data";
+import {clamp} from "../utils/math";
+import {NotificationContext} from "./NotificationContextProvider";
+import {
+    updateWindowUrlHashParams,
+    URL_HASH_PARAMS_DEFAULT,
+    URL_SEARCH_PARAMS_DEFAULT,
+    UrlContext,
+} from "./UrlContextProvider";
+
+
+const StateContext = createContext<null>(null);
```

Comment:
Sorry for missing this in the previous review - we should no longer need to create a context here. And we should remove the unused `export {StateContext}`. instead, we can simply
```
return <div>
    {children}
</div>
```
in AppController.

Also, we should rename `StateContextProviderProps` -> `AppController`.

@coderabbitai create an issue to track this